### PR TITLE
(feat) support singleAttributePerLine

### PR DESCRIPTION
**Pull request by david-ding**
This PR addresses https://github.com/sveltejs/prettier-plugin-svelte/issues/305 by adding support for the `singleAttributePerLine` option from prettier

### CODE REVIEW

File: src/lib/extractAttributes.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { AttributeNode, TextNode } from '../print/nodes';
 
 export function extractAttributes(html: string): AttributeNode[] {
-    const extractAttributesRegex = /<[a-z]+\s*(.*?)>/i;
-    const attributeRegex = /([^\s=]+)(?:=("|')(.*?)\2)?/gi;
+    const extractAttributesRegex = /<[a-z]+[\s\n]*([\s\S]*?)>/im;
+    const attributeRegex = /([^\s=]+)(?:=("|')([\s\S]*?)\2)?/gim;
```
Comment (dummdidumm) [inline src/lib/extractAttributes.ts:5]: What's the reason for this change?

File: src/print/helpers.ts [hunk 2]
```diff
@@ -48,3 +65,59 @@ export function replaceEndOfLineWith(text: string, replacement: Doc) {
...
+        | ScriptNode
+        | BodyNode
+        | OptionsNode
+        | SlotTemplateNode,
+    options: ParserOptions,
+    print: PrintFn,
+): PrintFn {
+    return (path) =>
+        path.getNode().name !== snippedTagContentAttribute
+            ? [getAttributeLine(node, options), path.call(print)]
```
Comment (dummdidumm) [inline src/print/helpers.ts:121]: We still need to use `concat` here even if it's now deprecated because we support Prettier versions where that wasn't a thing yet.

**Review by dummdidumm — COMMENTED** [review]
Thank you! This mostly looks good, except the tests are currently failing. Also left two comments.

File: src/print/helpers.ts (hunk 2 above)
Comment (david-ding) [reply src/print/helpers.ts:121]: > Thank you! This mostly looks good, except the tests are currently failing. Also left two comments.

Thanks! I'm also not sure why the tests are failing, I ran `npm test` and they all passed locally 😕

**Comment by dummdidumm** [comment]
For the tests to pass you probably need to bump Prettier in the devDeps and commit that package-lock change - it probably uses a version where that new option wasn't present yet.

**Comment by david-ding** [comment]
oh I know what happened, I used `yarn` initially so it didn't use `package-lock.json`, now I can see the tests failing, will fix soon

**Comment by aradalvand** [comment]
@dummdidumm Sorry if this is a dumb question but when will this be released? Thanks.

**Comment by dummdidumm** [comment]
Don't bump issues in separate places. I saw that, and I'll get round to it eventually.
